### ISSUE TITLE
Enable switching from Independent to Mentored Mode

### ISSUE THREAD

**Issue by iHiD** [exercism/exercism.io#3985]
We don't want someone to go through and be able to overload mentors with the same problematic code by not learning as they go. So what should happen? Should a user be penalised for switching backwards and forth? Should we just not worry about it unless it becomes a problem?

So after discussion internally, our working proposal is:
- [x] Someone can switch back from independent mode to normal mode in code
- [x] We add a "completed unapproved" visual state to the core-track exercises so you can see that.
- [x] Someone can switch back from independent mode to normal mode in UI
- [ ] Any solutions that they've worked on in independent mode are "completed unapproved"
- [ ] Relevant side exercises are not unlocked.
- [x] Learners can choose to cancel the "completed" status and put the exercise back into mentoring mode (similar to how this works for legacy solutions). 
- [x] Only one exercise can be put back into "unapproved" mode at a time (we'll make this same change for legacy so as not to overwhelm mentors).

**Comment by iHiD** [comment]
So after discussion internally, our working proposal is:
- Someone can switch back from independent mode to normal mode
- Any solutions that they've worked on in independent mode are "completed unapproved"
- Relevant side exercises are not unlocked.
- We add a "completed unapproved" visual state to the core-track exercises so you can see that.
- Learners can choose to cancel the "completed" status and put the exercise back into mentoring mode (similar to how this works for legacy solutions). 
- Only one exercise can be put back into "unapproved" mode at a time (we'll make this same change for legacy so as not to overwhelm mentors).

**Comment by nicolechalmers** [comment]
See Hello World for example of exercise attempted in independent mode.

I'm not sure if the tick/check should be grey or green. I think probably green so it feels official if the student is happy the solution is complete but doesn't want to bother with mentoring.

@iHiD The un-mentored icon is font awesome user-alt-slash

<img width="691" alt="screen shot 2018-07-18 at 13 06 04" src="https://user-images.githubusercontent.com/5421823/42880203-6cb86b56-8a8b-11e8-83e3-680eb1368e2a.png">

**Comment by kytrinyx** [comment]
I think this sounds like a good starting point.

The one question I have is this:

> Relevant side exercises are not unlocked.

How _would_ you unlock those side exercises? Is this by resubmitting the exercise for mentorship later and then getting it approved? Would this also apply to people who complete without approval in Normal mode?

**Comment by zancas** [comment]
This is relevant to me!   After waiting for mentor feedback for several days I switched to independent mode.   Now I do not see how to switch back to mentored mode.

**Comment by Ornataweaver** [comment]
@zancas Me neither.

**Comment by Ornataweaver** [comment]
I can't turn on the MENTOR MODE.
Or the way to doing so is so difficult that I couldn't find it after 15 mins.

**Comment by iHiD** [comment]
@Ornataweaver

This issue isn't people complaining about not being able to switch to mentored mode. It's a proposal for that feature. If you have suggestions not covered in the discussion please feel free to suggest them, but "Me too" and complaining posts on existing issues hinder rather than help.

**Comment by c-dante** [comment]
Would it make sense for a feature to get feedback on one problem while in independent mode?

Something like, "Request Mentor Feedback" for an individual problem, which keeps you in independent but opens up the one solution for feedback.

Unrelated but related, I've found myself in a situation with the rust track where there are very very few mentors, so I've done all the possible challenges. I don't want to switch to independent mode because I want feedback (I'm still learning the language itself, and it's very nice to see solutions/feedback on misuse or bad patterns), but I'm also at a loss for moving forward.

**Comment by zancas** [comment]
@c-dante I am in an identical situation on the Rust track.   I am effectively blocked from using exercism.

**Comment by zancas** [comment]
@kytrinyx @iHiD I am very enthusiastic about this project!  I believe that you're fulfilling a critical and growing need.  Thank you! 
The current policy forces motivated learners to either stop, or switch to independent mode.  It's very demoralizing.
If there's some way I can help please let me know.

**Comment by zancas** [comment]
@iHiD @kytrinyx I reread the scenario you are concerned about...   I will call it the "Unmentor-able Prolific User" scenario
 I think it's very important to factor in another possible scenario (because it's quite close to my own).
 
Scenario:

    A very motviated User has completed all possible exercises without mentor feedback.

    In the current system, since the User has no control over when a mentor will review their problem they are faced with an unpleasant choice, either stop working on the track and wait an unknown amount of time for feedback, or permanently forgo the benefits of mentorship in order to continue getting the benefits of well-designed exercises.

     This is _more_ likely, the _more_ motivated the User is, so the current policy is particularly biased _against_ the most motivated Users.

    The policy also is also biased against language tracks with fewer mentors, since those are the tracks most likely to block on review.  (I specifically chose exercism.io because relatively few resources exist for practicing Rust.)

     Also, please consider that Users who are uninterested in incorporating mentor feedback, i.e. updating their code in response to comments, have no real incentive to rejoin the mentor track, while those (like myself) intent on maximizing the benefits of exercism are highly motivated to update their solutions with mentor feedback.

   In short, I think disallowing Users from rejoining the "mentored" track certainly has significant downsides (for me, and Users like me), and potentially has no benefit (since we're not certain the "Unmentor-able Prolific User" is an important factor).

   Please make it possible for me to return to the "mentored" track.

**Comment by Ornataweaver** [comment]
@iHiD Sorry for my bad tone.
When I clicked on independent mode, I remember "you will be able to join back to mentor mode". Maybe my mind made that up since I couldn't find that statement on the website later. Mind creates what it expects! :D
That's why I was surprised when I read this topic. I surely understood it, but since I remembered that statement, I was confused and I was seeing a contradiction.

**Comment by Ornataweaver** [comment]
Also the idea of clicking on something, and not being able to undo it on a website is strange to me.
That's why my mind expected there to be a way to turn MENTOR MODE on again.

**Comment by Ornataweaver** [comment]
Check https://github.com/exercism/exercism.io/issues/3810, Other people are surprised like me.
